MetaLamp/nft-marketplace/client/src/Component/CreateBundleForm.purs

```diff
@@ -0,0 +1,353 @@
+      , HH.div
+          [ class_ "control" ]
+          [ UI.buttonPrimary
+              [ HE.onClick \_ -> Just $ F.injAction SubmitAll ]
+              [ HH.text "Submit" ]
+          ]
+      ]
+    where
+    mkCategoryForm i = do
+      let
+        handler = Just <<< F.injAction <<< HandleCategoryForm i
+      HH.slot _categoryForm i categoryFormComponent unit handler
+
+    mkIpfsCidForm i = do
+      let
+        handler = Just <<< F.injAction <<< HandleIpfsCidForm i
+      HH.slot _ipfsCidForm i ipfsCidFormComponent unit handler
+
+    _name = SProxy :: SProxy "name"
+
+    _description = SProxy :: SProxy "description"
+
+    _categoryForm = SProxy :: SProxy "categoryForm"
+
+    _ipfsCidForm = SProxy :: SProxy "ipfsCidForm"
+
+    _revealIssuer = SProxy :: SProxy "revealIssuer"
+
+-----
+-- Category form, nested inside
+-----
+-- Form types
+type CategoryInfo
+  = { | CategoryRow F.OutputType }
+
+newtype CategoryForm r f
+  = CategoryForm (r (CategoryRow f))
+
+derive instance newtypeCategoryForm :: Newtype (CategoryForm r f) _
+
+type CategoryRow f
+  = ( category :: f Void String String
+    )
+
+-- Form component types
+type CategoryFormSlot
+  = H.Slot (F.Query' CategoryForm) CategoryFormMessage
+
+data CategoryFormAction
+  = RemoveMe1
+
+data CategoryFormMessage
+  = Destroy1
+
+-- Form spec
+categoryFormComponent ::
+  forall m.
+  MonadAff m => F.Component CategoryForm (Const Void) () Unit CategoryFormMessage m
+categoryFormComponent =
+  F.component (const categoryFormInput)
+    $ F.defaultSpec
+        { render = render
+        , handleAction = handleAction
+        }
+  where
+  categoryFormInput :: F.Input' CategoryForm m
+  categoryFormInput =
+    { validators:
+        CategoryForm
+          { category: F.noValidation
+          }
+    , initialInputs: Nothing
+    }
+
+  handleAction = case _ of
+    RemoveMe1 -> H.raise Destroy1
+
+  render st =
+    UI.formContent_
+      [ HH.div
+          [ class_ "field" ]
+          [ UI.buttonPrimary
+              [ HE.onClick \_ -> Just $ F.injAction RemoveMe1 ]
+              [ HH.text "Remove Category" ]
+          ]
+      , UI.input
+          { label: "Inner category"
+          , help: Right "Provide any additional category you'd like."
+          , placeholder: "photos"
+          }
+          [ HP.value $ F.getInput _category st.form
+          , HE.onValueInput $ Just <<< F.set _category
+          ]
+      ]
+    where
+    _category = SProxy :: SProxy "category"
+
+-----
+-- IpfsCid form, nested inside
+-----
+-- Form types
+type IpfsCidInfo
+  = { | IpfsCidRow F.OutputType }
+
+newtype IpfsCidForm r f
+  = IpfsCidForm (r (IpfsCidRow f))
+
+derive instance newtypeIpfsCidForm :: Newtype (IpfsCidForm r f) _
+
+type IpfsCidRow f
+  = ( ipfsCid :: f Void String String
+    )
+
+-- Form component types
+type IpfsCidFormSlot
+  = H.Slot (F.Query' IpfsCidForm) IpfsCidFormMessage
+
+data IpfsCidFormAction
+  = RemoveMe2
```

Comment:
`RemoveCategory` and `RemoveIpfsCid` will be more informative, or maybe we can split `categoryFormComponent` and `ipfsCidFormComponent` by different files and use `RemoveMe`, `DestroyMe`

Reply:
I don't think we should split components for now if they are not reused

Reply:
Category is used two times, but that is still arguably low